### ISSUE TITLE
Problem parsing C++ declaration with char default argument

### ISSUE THREAD

**Issue by randombit** [sphinx-doc/sphinx#5147]
Subject: Parsing a C++ function with a char argument with default value fails.

### Problem

The following declaration

```
.. cpp:function:: std::string generate_bcrypt(const std::string& password, \
                    RandomNumberGenerator& rng, \
                    uint16_t work_factor = 10, \
                    char bcrypt_version = 'a')
```

cannot be parsed.

#### Error logs / results
```
Error in postfix expression, expected primary expression or type.
If primary expression:
  Invalid definition: Expected identifier in nested name. [error at 185]
    std::string generate_bcrypt(const std::string& password,                  RandomNumberGenerator& rng,                  uint16_t work_factor = 10,                  char bcrypt_version = 'a')
    -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------^
If type:
  Invalid definition: Expected identifier in nested name. [error at 185]
    std::string generate_bcrypt(const std::string& password,                  RandomNumberGenerator& rng,                  uint16_t work_factor = 10,                  char bcrypt_version = 'a')
    -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------^

```

#### Expected results

I would expect this to parse. Changing the quotes to "a" avoids the problem, but unfortunately means the C++ syntax is not correct. [Single `char` is 'a', with "a" instead meaning a NULL terminated string]

### Environment info
- OS: Arch Linux
- Python version: 3.6.5
- Sphinx version: 1.7.5

**Comment by jakobandersen** [comment]
Thanks for reporting. The problems is that character literals are simply not yet supported (https://github.com/sphinx-doc/sphinx/blob/master/sphinx/domains/cpp.py#L4181). We should be able to get that fixed for 1.8.

**Comment by jakobandersen** [comment]
This should work in the upcoming version 1.8.
